### ISSUE TITLE
.htaccess mod no longer needed?

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#1608]
Currently, the /public/.htaccess file has the section that reads:

`# if in subdir comment above line, uncomment below one and replace <OSPOS path> with your path`
`# RewriteRule ^(.*)$ /<OSPOS path>/public/index.php?/$1 [L]`

I went to rework this so that it didn't matter if you were in a subdirectory or a webroot directory and it appears that the code is obsolete.  I'm guessing that with my changes to the .htaccess file in the root to redirect to the public folder, it makes this change unnecessary.  I tested two different instances (both in subdirectories) without the code above enabled using the latest bintray and it worked super.  If, someone can confirm that this is not needed on their instance (that resides in a subdirectory) then I will submit a pull request to just remove the code.

**Comment by objecttothis** [comment]
also, application/helpers/sale_helper.php is empty.  Is it needed or can it be deleted?

**Comment by daN4cat** [comment]
I would say no, I would not rely on an additional redirection to fix a wrong .htaccess configuration. So please don't change that.

**Comment by daN4cat** [comment]
Yes sale_helper.php is empty but I never removed it because we might need it in some refactoring exercise. That's not an urgent thing to sort out in my opinion.

**Comment by objecttothis** [comment]
OK, that's fine.  I would like to rework the /public/.htaccess so that the mod_rewrite accounts for both builds in the webroot and in subdirectories.  This would prevent users from needing to modify the .htaccess depending on where their build is stored.  I think I would just modify it so that it pulled the build location from the request URI dynamically.  This also reduces support requests that are along the lines of "it's not working, I'm in a subdirectory, what do I do?"

**Comment by daN4cat** [comment]
Yes, if there is a smart way to do that please go ahead. I followed what was there.
Also bear in mind that the trick hides the index.php from the URL and etc.

**Comment by objecttothis** [comment]
OK, can you assign this to me?

**Comment by daN4cat** [comment]
@objecttothis can we close this?

**Comment by objecttothis** [comment]
Let me take a crack at it Monday morning to combine the two elements.

**Comment by objecttothis** [comment]
Since I am going on sabbatical, I am going to punt this to 4.0.0.  It's not that involved of a mod, but I need to triage the more important things.

**Comment by daN4cat** [comment]
oh good.
